apps/src/code-studio/components/aiTutor/aiTutorPanel.jsx

```diff
@@ -1,29 +1,94 @@
-import React from 'react';
+import React, {useState} from 'react';
 import PropTypes from 'prop-types';
+import {useDispatch} from 'react-redux';
 import i18n from '@cdo/locale';
 import style from './ai-tutor.module.scss';
+import Checkbox from '@cdo/apps/componentLibrary/checkbox';
 import AITutorPanelContainer from '@cdo/apps/code-studio/components/aiTutor/aiTutorPanelContainer';
-import CompilationAssistant from './compilationAssistant';
+import CompilationTutor from './compilationTutor';
+import ValidationTutor from './validationTutor';
+import GeneralChatTutor from './generalChatTutor';
+import {addAIResponse} from '@cdo/apps/aiTutor/redux/aiTutorRedux';
 const icon = require('@cdo/static/ai-bot.png');
 
-export default class AITutorPanel extends React.Component {
-  static propTypes = {
-    level: PropTypes.shape({
-      id: PropTypes.number.isRequired,
-      type: PropTypes.string.isRequired,
-    }),
-  };
+const AITutorPanel = props => {
+  const dispatch = useDispatch();
+  const {level} = props;
```

Comment:
nit: You could destructure this in props, i.e. `const AITutorPanel = ({level}) => { ... })`.